src/recipe/recipe.controller.ts

```diff
@@ -58,37 +58,29 @@ export class RecipeController {
     return await this.recipeService.updateRecipe(Number(recipeId), newRecipe);
   }
 
-  @Get('/id/:id')
-  @ApiOperation({ summary: 'Get one recipe by id' })
-  @ApiResponse({ type: RecipeResponse })
-  @ApiParam({ name: 'id', required: true })
-  async getOneRecipe(@Param('id') recipeId: number): Promise<RecipeResponse> {
-    return await this.recipeService.findRecipeById(Number(recipeId));
-  }
-
-  @Get('/list')
-  @UseGuards(AuthGuard('jwt'), IsAdminGuard)
+  @Get()
+  @UseGuards(AuthGuard('jwt'))
   @ApiOperation({ summary: 'Get all recipes' })
   @ApiResponse({ type: [RecipeResponse] })
   @ApiUnauthorizedResponse({ description: 'Not logged in' })
   @ApiForbiddenResponse({
     description: 'User is not an admin',
   })
   async getAllRecipes(
-    @Query('page') page: string,
-    @Query('limit') limit: string,
+    @Query() query: QueryParameters,
   ): Promise<RecipeResponse[]> {
-    return await this.recipeService.findAllRecipes(Number(limit), Number(page));
+    return await this.recipeService.findAllRecipes(
+      query.limit,
+      query.page,
+      query.name,
+    );
   }
 
-  @Get('/name')
-  @UseGuards(AuthGuard('jwt'))
-  @ApiOperation({ summary: 'Get a recipe by name' })
-  @ApiResponse({ type: [RecipeResponse] })
-  @ApiUnauthorizedResponse({ description: 'Not logged in' })
-  async getRecipeByName(
-    @Query('query') query: string,
-  ): Promise<RecipeResponse[]> {
-    return await this.recipeService.findRecipeByName(query);
+  @Get('/:id')
+  @ApiOperation({ summary: 'Get one recipe by id' })
+  @ApiResponse({ type: RecipeResponse })
+  @ApiParam({ name: 'id', required: true })
+  async getOneRecipe(@Param('id') recipeId: number): Promise<RecipeResponse> {
```

Comment:
this param is also an input - should be validate. Try using validation pipe https://docs.nestjs.com/techniques/validation
Then you would not need to do: 
`Number(recipeId)`